### PR TITLE
fix: Account for the way the mobile app represents number selections (M2-8174)

### PR DESCRIPTION
**Pull request by sultanofcardio**
### 📝 Description

🔗 [Jira Ticket M2-8174](https://mindlogger.atlassian.net/browse/M2-8174)

This PR updates the T score calculation to account for the way the mobile app represents number selections. 

The mobile app will submit the answer to the age question as a number selection when the admin configures the field as a dropdown. However, its representation of this data differs from the web app: the mobile app will use a string, whereas the web app uses a number. Therefore, we need to account for both in the report generation.

### 📸 Screenshots

N/A

### 🪤 Peer Testing

1. Create an activity with at least one scorable item
2. Create a subscale with a lookup table containing severity information
    <details>
      <summary>Here's a sample one</summary>
      
      ```csv
      score,rawScore,age,sex,optionalText,severity
     10,1,15,M,https://gist.githubusercontent.com/benbalter/3914310/raw/f757a33411082da23f0ad4a124b45fcdacc1b43f/Example--text.txt,Minimal
      20,2,15,M,https://gist.githubusercontent.com/benbalter/3914310/raw/f757a33411082da23f0ad4a124b45fcdacc1b43f/Example--text.txt,Mild
      30,3,15,M,Markdown Text Here,Moderate
      40,4,15,F,Good,Severe
      50,5,15,F,Nice,Severe
      ```
    </details>
3. Switch the age field type to dropdown in the subscale settings
4. Create a report score that uses scoring type "Score" and link it to the subscale from earlier
5. Add a score condition to the report score that relies on the score being equal to one from the `score` column of the lookup table (e.g. score is equal to 30)
6. Save and complete the activity via the mobile app. Complete the activity such that the total score will match the entry in the `rawScore` column of the lookup table in the same row as the score from the score condition (e.g. the total score will be 3, and you will select Male and 15 for the sex and age respectively, which will map to the 30 from above)
7. Download the latest PDF report for the participant in the activity
8. Verify that the message from the score condition is present

### ✏️ Notes

I haven't been able to verify this locally. My reports keep coming up empty for what I think is an unrelated reason. Nevertheless, I wanted to get this into more hands so that it can be reviewed

### CODE REVIEW

**Review by rcmerlo — DISMISSED** [review]

File: src/models/activity.ts [hunk 1]
```diff
@@ -125,7 +125,7 @@ export class ActivityEntity {
             let reportedAge: number | null = null
             if (typeof ageAnswer === 'string') {
               reportedAge = Number(ageAnswer)
-            } else if ('value' in ageAnswer && typeof ageAnswer.value === 'number') {
+            } else if ('value' in ageAnswer && ['number', 'string'].includes(typeof ageAnswer.value)) {
```
Comment (rcmerlo) [inline src/models/activity.ts:128]: Should convert ageAnswer.value to Number when it is a string

**Review by rcmerlo — CHANGES_REQUESTED** [review]

File: src/models/activity.ts (hunk 1 above)
Comment (sultanofcardio) [reply src/models/activity.ts:128]: Hey Rodrigo. I don't think I understand the reasoning for this suggestion. Why should we convert it to a number?

Comment (rcmerlo) [reply src/models/activity.ts:128]: Comparison bellow in line 132 will fail and reportAge variable is a Number or null

Comment (sultanofcardio) [reply src/models/activity.ts:128]: Oh, I realise the code here is doing something slightly different than the admin (it's also missing the age range check). Thanks for pointing this out, I'll update it

File: src/models/activity.ts [hunk 2]
```diff
@@ -122,14 +122,27 @@ export class ActivityEntity {
...
-              reportedAge = Number(ageAnswer)
-            } else if ('value' in ageAnswer && typeof ageAnswer.value === 'number') {
-              reportedAge = ageAnswer.value
+            let reportedAge: string | null = null
+            if (
+              typeof ageAnswer === 'string' ||
+              ('value' in ageAnswer && ['number', 'string'].includes(typeof ageAnswer.value))
+            ) {
+              reportedAge = String(ageAnswer)
+            }
```
Comment (mbanting) [inline src/models/activity.ts:131]: `reportedAge = String(ageAnswer)` won't work if `ageAnswer` is the dropdown value object and the age is in  `ageAnswer.value`.  The following should fix it:
```suggestion
            if (typeof ageAnswer === 'string') {
              reportedAge = ageAnswer
            } else if (ageAnswer && 'value' in ageAnswer && ['number', 'string'].includes(typeof ageAnswer.value)) {
              reportedAge = String(ageAnswer.value)
            }
```

**Review by mbanting — CHANGES_REQUESTED** [review]
Tested and unfortunately the lookup score wasn't displayed when age is rendered as a dropdown (on either web or mobile). Added a comment on where the issue likely is and a potential fix.

File: src/models/activity.ts (hunk 2 above)
Comment (sultanofcardio) [reply src/models/activity.ts:131]: Thanks, this is actually what I meant to do 🤦‍♂️

**Review by mbanting — APPROVED** [review]

**Pull request merged**